### PR TITLE
Add localfile wildcard support for windows IT

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|-------------|
| #3786     |

## Description

This PR adds test support for wildcards usage in the localfile `location` tag for Windows agents.

### Added

- Added 6 test cases in `/test_logcollector/test_location_wildcards` suite. Testing different wildcards combinations and a case where wildcards do not match a valid file

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @Deblintrake09  (Developer)  | test_logcollector           |   | [:green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh-qa/files/11056100/Manager_Results.zip)  |  Manager        |   https://github.com/wazuh/wazuh-qa/pull/3906/commits/1536dd807b366db964ff43ce30ef77e96e808444      | Nothing to highlight |
| @Deblintrake09  (Developer)  | test_logcollector           |   | [:green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh-qa/files/11055296/LinuxAgent_Results.zip) |  Linux agent        |   https://github.com/wazuh/wazuh-qa/pull/3906/commits/1536dd807b366db964ff43ce30ef77e96e808444      | Nothing to highlight |
| @Deblintrake09  (Developer)  | test_logcollector           |   | [:green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh-qa/files/11055096/WinAgent_Results.zip)  |  Windows agent        |   https://github.com/wazuh/wazuh-qa/pull/3906/commits/1536dd807b366db964ff43ce30ef77e96e808444      | Nothing to highlight |
| @damarisg  (Reviewer)   | test_logcollector   | [🟢](https://ci.wazuh.info/job/Test_integration/38028/)  | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |Manager         |   https://github.com/wazuh/wazuh-qa/pull/3906/commits/ae13d712687903ffaf77a53ca25e689293f45e8f | Nothing to highlight |
| @damarisg  (Reviewer)   | test_logcollector    | [🟢](https://ci.wazuh.info/job/Test_integration/38029/)  | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | Linux agent         |  https://github.com/wazuh/wazuh-qa/pull/3906/commits/ae13d712687903ffaf77a53ca25e689293f45e8f | Nothing to highlight |
| @damarisg  (Reviewer)   | test_logcollector    | [🟢](https://ci.wazuh.info/job/Test_integration/38030/)  | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |  Windows agent | https://github.com/wazuh/wazuh-qa/pull/3906/commits/ae13d712687903ffaf77a53ca25e689293f45e8f      | Nothing to highlight |
| @damarisg  (Reviewer)   | test_logcollector    | [🟢](https://ci.wazuh.info/view/Tests/job/Test_integration/38062/)  | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | Solaris and macOS agent         |  https://github.com/wazuh/wazuh-qa/pull/3906/commits/ae13d712687903ffaf77a53ca25e689293f45e8f | Nothing to highlight |
| @Deblintrake09  (Developer)  | test_logcollector           |   | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/11402269/Manager.zip)  |  Manager        |   https://github.com/wazuh/wazuh-qa/pull/3906/commits/5de9cca5c1d333098c942e9902d8aab8ec17f840      | Additional results after moving test to new folder |
| @Deblintrake09  (Developer)  | test_logcollector           |   | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/11402268/LinuxAgent.zip) |  Linux agent        |   https://github.com/wazuh/wazuh-qa/pull/3906/commits/5de9cca5c1d333098c942e9902d8aab8ec17f840      | Additional results after moving test to new folder |
| @Deblintrake09  (Developer)  | test_logcollector           |   | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/11402270/WinAgent.zip)  |  Windows agent        |   https://github.com/wazuh/wazuh-qa/pull/3906/commits/5de9cca5c1d333098c942e9902d8aab8ec17f840      | Additional results after moving test to new folder |

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Release report: TBD
 
 Wazuh commit: TBD \
 Release report: TBD
-
+- Add Windows location wildcards tests ([#4263](https://github.com/wazuh/wazuh-qa/pull/4263)) \- (Tests + Framework)
```
Comment (juliamagan) [inline CHANGELOG.md:14]: This line should be in 4.6.0

File: tests/integration/test_logcollector/test_location/test_win_location_wildcards
Comment (juliamagan) [inline tests/integration/test_logcollector/test_location/test_win_location_wildcards:1]: Missing `.py` in name

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: tests/integration/test_logcollector/test_location/test_win_location_wildcards
Comment (Deblintrake09) [reply tests/integration/test_logcollector/test_location/test_win_location_wildcards:1]: Done in https://github.com/wazuh/wazuh-qa/pull/4263/commits/6454c401577aed8fee779b0bb2e84d41982708a2

File: CHANGELOG.md (hunk 1 above)
Comment (Deblintrake09) [reply CHANGELOG.md]: Done in https://github.com/wazuh/wazuh-qa/pull/4263/commits/6454c401577aed8fee779b0bb2e84d41982708a2

**Review by juliamagan — APPROVED** [review]
LGTM

File: tests/integration/test_logcollector/test_configuration/test_basic_configuration_ignore_binaries.py [hunk 2]
```diff
@@ -64,7 +64,9 @@
 from wazuh_testing.tools.services import control_service
 from wazuh_testing.tools.file import truncate_file
 import wazuh_testing.api as api
-from wazuh_testing.tools.monitoring import LOG_COLLECTOR_DETECTOR_PREFIX, WINDOWS_AGENT_DETECTOR_PREFIX, FileMonitor
+from wazuh_testing.tools.monitoring import FileMonitor
+from wazuh_testing.modules.logcollector import LOG_COLLECTOR_PREFIX, WINDOWS_AGENT_PREFIX
```
Comment (jnasselle) [inline tests/integration/test_logcollector/test_configuration/test_basic_configuration_ignore_binaries.py:68]: These constants renaming were not populated in the code. See:
https://github.com/wazuh/wazuh-qa/pull/4263/files#diff-bb0af03f6d265932e007df345ab06f3bfa0c756c679b388667e4f7cf4c507868R88
https://github.com/wazuh/wazuh-qa/pull/4263/files#diff-bb0af03f6d265932e007df345ab06f3bfa0c756c679b388667e4f7cf4c507868R91

**Review by jnasselle — CHANGES_REQUESTED** [review]
Also it seems to be some problems loading configurations.

```
============================= test session starts ==============================
platform linux -- Python 3.10.9, pytest-7.1.2, pluggy-1.0.0 -- /bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.10.9', 'Platform': 'Linux-3.10.0-1160.81.1.el7.x86_64-x86_64-with-glibc2.17', 'Packages': {'pytest': '7.1.2', 'pluggy': '1.0.0'}, 'Plugins': {'testinfra': '5.0.0', 'metadata': 
'2.0.4', 'html': '3.1.1'}}
rootdir: /tmp/Test_integration_B41699_20230726122239/tests/integration, configfile: pytest.ini
plugins: testinfra-5.0.0, metadata-2.0.4, html-3.1.1
collecting ... collected 631 items / 4 errors / 97 deselected / 534 selected

==================================== ERRORS ====================================
_ ERROR collecting test_logcollector/test_configuration/test_basic_configuration_ignore_binaries.py _
test_logcollector/test_configuration/test_basic_configuration_ignore_binaries.py:91: in <module>
    prefix = LOG_COLLECTOR_DETECTOR_PREFIX
E   NameError: name 'LOG_COLLECTOR_DETECTOR_PREFIX' is not defined
______ ERROR collecting test_logcollector/test_location/test_location.py _______
test_logcollector/test_location/test_location.py:187: in <module>
    configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
/usr/local/python-3.10/lib/python3.10/site-packages/wazuh_testing-4.6.0-py3.10.egg/wazuh_testing/tools/configuration.py:446: in load_wazuh_configurations
    with open(yaml_file_path) as stream:
E   FileNotFoundError: [Errno 2] No such file or directory: '/tmp/Test_integration_B41699_20230726122239/tests/integration/test_logcollector/test_location/data/configuration_template/configuration_loc
ation.yaml'
__ ERROR collecting test_logcollector/test_location/test_location_exclude.py ___
test_logcollector/test_location/test_location_exclude.py:187: in <module>
    configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
/usr/local/python-3.10/lib/python3.10/site-packages/wazuh_testing-4.6.0-py3.10.egg/wazuh_testing/tools/configuration.py:446: in load_wazuh_configurations
    with open(yaml_file_path) as stream:
E   FileNotFoundError: [Errno 2] No such file or directory: '/tmp/Test_integration_B41699_20230726122239/tests/integration/test_logcollector/test_location/data/configuration_template/configuration_loc
ation.yaml'
_ ERROR collecting test_logcollector/test_location/test_win_location_wildcards.py _
test_logcollector/test_location/test_win_location_wildcards.py:69: in <module>
    configurations = load_configuration_template(configurations_path, configuration_parameters,
/usr/local/python-3.10/lib/python3.10/site-packages/wazuh_testing-4.6.0-py3.10.egg/wazuh_testing/tools/configuration.py:696: in load_configuration_template
    configuration = file.read_yaml(data_file_path)
/usr/local/python-3.10/lib/python3.10/site-packages/wazuh_testing-4.6.0-py3.10.egg/wazuh_testing/tools/file.py:61: in read_yaml
    with open(file_path) as f:
E   FileNotFoundError: [Errno 2] No such file or directory: '/tmp/Test_integration_B41699_20230726122239/tests/integration/test_logcollector/test_location/data/configuration_template/configuration_win
_location_wildcards.yaml'
- generated html file: file:///tmp/Test_integration_B41699_20230726122239/report.html -
=========================== short test summary info ============================
ERROR test_logcollector/test_configuration/test_basic_configuration_ignore_binaries.py
ERROR test_logcollector/test_location/test_location.py - FileNotFoundError: [...
ERROR test_logcollector/test_location/test_location_exclude.py - FileNotFound...
ERROR test_logcollector/test_location/test_win_location_wildcards.py - FileNo...
!!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!
======================= 97 deselected, 4 errors in 9.30s =======================
```

File: tests/integration/test_logcollector/test_configuration/test_basic_configuration_ignore_binaries.py (hunk 2 above)
Comment (Deblintrake09) [reply tests/integration/test_logcollector/test_configuration/test_basic_configuration_ignore_binaries.py:68]: - Applied fix in https://github.com/wazuh/wazuh-qa/pull/4263/commits/7249189978560208455c8e11bbf18a6023187be2

## Fix Evidence

| Local | Jenkins |
|:--|:--|
| [:green_circle:](https://github.com/wazuh/wazuh-qa/files/12177247/Logcollector.Local.zip) | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/12177249/manager_html_report_Test_integration_B41704_20230726211927.zip)

**Review by jnasselle — APPROVED** [review]

**Review by juliamagan — APPROVED** [review]
LGTM

**Pull request merged**